src/executeJar.ts

```diff
@@ -240,6 +241,63 @@ export async function loadContextsFromChangelogFile(
       reject(error as Error);
     }
   });
+
+  /**
+   * Sets the context for all files that point to the same changelog file.
+   *
+   * @param contexts - the contexts that were loaded from the changelog file
+   * @param changelogFile - the absolute path to the changelog file
+   */
+  function setContextForCache(contexts: string[], changelogFile: string): void {
+    readLiquibaseConfigurationNames()
+      .then((names) => {
+        if (!names) {
+          return;
+        }
+        for (const configurationName of names) {
+          getPathOfConfiguration(configurationName)
+            .then((propertyFile) => {
+              if (!propertyFile) {
+                return;
+              }
+
+              const data = readFullValues(configurationName, propertyFile, {
+                defaultDatabaseForConfiguration: getDefaultDatabaseForConfiguration(),
+                liquibaseDirectoryInProject: getLiquibaseFolder(),
+                customDrivers: getCustomDrivers(),
+              });
+
+              //here you have to use the absolute path to the changelog file
+              const previousSelectedContexts: ContextSelection = cacheHandler.readContexts(
+                propertyFile,
+                path.join(getWorkFolder(), data.changelogFile)
+              );
+
+              if (data.changelogFile.includes(path.relative(getWorkFolder(), changelogFile))) {
+                // save the loaded context into the cache
+                cacheHandler.saveContexts(propertyFile, changelogFile, {
```

Comment:
One thing I have noticed now:

Prerequisites:

- system.liquibase.properties pointing to system/changelog.xml
- data.liquibase.properties pointing to data/changelog.xml
- empy cache.json

Updating system.liquibase.properties with system/changelog.xml, loadedContexts and selectedContexts are stored fine.

Then using **data**.liquibase.properties with other changelog **system**/changelog.xml.
I still use "Load Contexts": After the loading, the loaded contexts are not stored for data.liquibase.properties, which I would expect.
The selected contexts are now fine.